test/unit/mdc-menu/simple.foundation.test.js

```diff
@@ -62,216 +94,358 @@ test('exports numbers', () => {
-    'setPosition', 'getAccurateTime',
+    'setPosition', 'setMaxHeight',
   ]);
 });
 
 test('#init throws error when the root class is not present', () => {
   const mockAdapter = td.object(MDCSimpleMenuFoundation.defaultAdapter);
-  td.when(mockAdapter.hasClass('mdc-simple-menu')).thenReturn(false);
+  td.when(mockAdapter.hasClass(cssClasses.ROOT)).thenReturn(false);
 
   const foundation = new MDCSimpleMenuFoundation(mockAdapter);
   assert.throws(() => foundation.init());
 });
 
 test('#init throws error when the necessary DOM is not present', () => {
   const mockAdapter = td.object(MDCSimpleMenuFoundation.defaultAdapter);
-  td.when(mockAdapter.hasClass('mdc-simple-menu')).thenReturn(true);
+  td.when(mockAdapter.hasClass(cssClasses.ROOT)).thenReturn(true);
   td.when(mockAdapter.hasNecessaryDom()).thenReturn(false);
 
   const foundation = new MDCSimpleMenuFoundation(mockAdapter);
   assert.throws(() => foundation.init());
 });
 
-testFoundation('#open adds the animation class to start an animation', ({foundation, mockAdapter, mockRaf}) => {
-  foundation.open();
-  mockRaf.flush();
-  mockRaf.flush();
-  td.verify(mockAdapter.addClass('mdc-simple-menu--animating'));
-});
+testFoundation('#open adds the animation class to start an animation',
+  ({foundation, mockAdapter}) => {
+    foundation.open();
+    td.verify(mockAdapter.addClass(cssClasses.ANIMATING_OPEN));
+  });
 
 testFoundation('#open adds the open class to the menu', ({foundation, mockAdapter, mockRaf}) => {
-  td.when(mockAdapter.hasClass('mdc-simple-menu--open-from-bottom-right')).thenReturn(true);
-
   foundation.open();
   mockRaf.flush();
   mockRaf.flush();
-  td.verify(mockAdapter.addClass('mdc-simple-menu--open'));
+  td.verify(mockAdapter.addClass(cssClasses.OPEN));
 });
 
 testFoundation('#open removes the animation class at the end of the animation',
   ({foundation, mockAdapter, mockRaf}) => {
-    td.when(mockAdapter.getAccurateTime()).thenReturn(0);
-    td.when(mockAdapter.hasClass('mdc-simple-menu--open-from-top-right')).thenReturn(true);
-
+    const clock = lolex.install();
     foundation.open();
+    td.verify(mockAdapter.addClass(cssClasses.ANIMATING_OPEN));
     mockRaf.flush();
+    td.verify(mockAdapter.addClass(cssClasses.OPEN));
+    clock.tick(numbers.TRANSITION_OPEN_DURATION);
     mockRaf.flush();
-    td.verify(mockAdapter.addClass('mdc-simple-menu--animating'));
-
-    td.when(mockAdapter.getAccurateTime()).thenReturn(500);
-    mockRaf.flush();
-    mockRaf.flush();
-    td.verify(mockAdapter.removeClass('mdc-simple-menu--animating'));
+    td.verify(mockAdapter.removeClass(cssClasses.ANIMATING_OPEN));
   });
 
 testFoundation('#open focuses the menu at the end of the animation', ({foundation, mockAdapter, mockRaf}) => {
-  td.when(mockAdapter.getAccurateTime()).thenReturn(0);
-
   foundation.open();
   mockRaf.flush();
-  mockRaf.flush();
-
-  td.when(mockAdapter.getAccurateTime()).thenReturn(500);
```

Comment:
Given that this test is only waiting for the very next frame, the test name seems inaccurate - are we actually focusing before the animation finishes? (That might be okay, but we should probably remove that part from the test name.)

Reply:
done.